peer/connector_test.go

```diff
@@ -0,0 +1,41 @@
+package peer
+
+import (
+	"testing"
+
+	"github.com/bitmark-inc/bitmarkd/peer/mocks"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+)
+
+func newTestConnector() *connector {
+	return &connector{}
+}
+
+func newTestMockUpstream(t *testing.T) (*gomock.Controller, *mocks.MockUpstreamIntf) {
+	ctl := gomock.NewController(t)
+	return ctl, mocks.NewMockUpstreamIntf(ctl)
+}
+
+func TestNextState(t *testing.T) {
+	c := newTestConnector()
+	orig := c.state
+	c.nextState()
```

Comment:
Why do we need to test the private function?

Reply:
usually not, but in connector has no other public test to cover this method

Reply:
I concern about we try to test the internal/private method by putting the test file in the same package with source file. It seems like a trick.

Reply:
I think it depends, for a reference of go source code https://github.com/golang/go/blob/master/src/crypto/sha512/sha512_test.go, it also put test with same package as target package

Reply:
I just took a look on your ref. It seems no private/internal method used for testing but i don't know why they put the test file and source file in the same package. Basically i am not sure we totally ignore all internal/private function from testing but in best practice we should only test the public function or the caller of internal/private function, that why i suggest you don't put the testing file and source file in the same package. If in rare case, you need to test the internal/private method, reflection or event public it as the method is only visible for testing is needed. Just in my experience, anyway if it took lot of time to change, you can keep your way, i just want to notice you.

Reply:
Reference code put test inside same package, so i think put test into same package is not that tricky.

Currently connector has no test, i would like to add some tests for better code protection, instead of leaving code without test.

I agree that internal method best not tested, but consider current situation of not a single test to Connector, i think it is worth violate some concept but actually provide higher code protection.

Reply:
OK